### DISCUSSION TITLE
How do we send email verification when user is not signed-in?

### DISCUSSION THREAD

**Discussion by anietieasuquo** [invertase/react-native-firebase#6292] [Q&A] [answered]
Hi there,

My app requires users to request email verification link if for some reason the original verification link cannot be used.

For this reason we would need to call `auth().user.sendEmailVerification()`. But in this case the user is not signed-in, so we do not have a user object.

We have tried to sign-in as anonymous user and update the email address to the user-provided email address, and then use it as `anonymousUser.sendEmailVerification()`, but we get the error `auth/email-already-in-use`.

So the question is; how do we send email verification link when the user is not signed-in?

**Comment by mikehardy** [comment, marked as answer]
Interesting - I haven't had this use case. Easiest way might be to use firebase-admin-sdk in a cloud function - have you tried that?

**Comment by anietieasuquo** [comment]
@mikehardy No I didn't try that, will try now.

**Reply by anietieasuquo** [reply]
Yeah that works 👍
